torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -328,61 +358,90 @@ uint64_t FusionExecutor::computeSharedMemory(
   return total;
 }
 
+uint64_t FusionExecutor::computeSharedMemory(
+    kir::ExpressionEvaluator& expr_eval,
+    const std::vector<const kir::Allocate*>& buffers,
+    const std::vector<glfdc::SymbolicExpr>& memoized,
+    bool align_padding,
+    uint64_t total) {
+  FUSER_PERF_SCOPE("computeSharedMemory::new");
+  size_t i = 0;
+  for (auto smem_alloc : buffers) {
+    if (smem_alloc->alias() == nullptr) {
+      auto mem_inferred = expr_eval.evaluate(memoized[i]);
+      TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
+          mem_inferred == expr_eval.evaluate(smem_alloc->size()),
+          "Wrong value calculated ",
+          mem_inferred.value(),
+          " vs ",
+          expr_eval.evaluate(smem_alloc->size()).value());
+      if (mem_inferred.has_value()) {
+        const uint64_t data_size = dataTypeSize(smem_alloc->buffer()->dtype());
+        // Add padding to align dynamic shared memory
+        if (align_padding) {
+          total = ceilDiv(total, data_size) * data_size;
+        }
+        total += mem_inferred.value() * data_size;
+      } else {
+        TORCH_INTERNAL_ASSERT(
+            false,
+            "Failed to evaluate the size ",
+            smem_alloc->size(),
+            " of shared memory buffer - T",
+            smem_alloc->buffer()->name());
+      }
+    }
+    ++i;
+  }
+  return total;
+}
+
 LaunchParams FusionExecutor::computeLaunchParams(
     const LaunchParams& launch_constraints,
     kir::ExpressionEvaluator& expr_eval) {
   FUSER_PERF_SCOPE("FusionExecutor::ComputeLaunchParams");
 
   LaunchParams launch_params;
+  const auto& kernel_summary = kernel()->summary();
 
   auto data_cache = compileTimeDataCache();
-
```

Comment:
I'm relatively open on this implementation and we can discuss more.

Putting these in either `KernelSummary` or `ExecutorCompileTimeEntry` wouldn't make much difference timing-wise: 

We discussed earlier that `ExecutorCompileTimeEntry` would be computed first time used while `KernelSummary` is computed in `GpuLower`. But generally we lower/compile these fusion graphs the first time we run it so the latency of the first graph launch will be combined with lowering and nvrtc compile either way. 

For the discussion, my thinking behind creating the `ExecutorCompileTimeEntry` while we already had `KernelSummary` was that I thought we'd need a caching space for pure performance improvement and the `data we put in the ExecutorCompileTimeCache` are:

1. Not particularly as "meaningful" as what we have in `KernelSummary`. They could be just some vector of pointers that we wouldn't pay particular attention to when lowering the graph. We just need them computed and cached so that we don't need to do it every time we run the kernels. 

2. and because of 1., some of them are subject to change in the future, possibly quite frequently, so the entry definitions were more ad hoc/flexible and I thought it'd be helpful to place the computation of the entries exactly in-place where they're needed.

But either way, we could always do follow up PR's if we think one way is particularly cleaner looking than the other.